components/datadog/agent/host.go

```diff
@@ -155,6 +162,27 @@ func (h *HostAgent) installIntegrationConfigsAndFiles(
 	allCommands := make([]*remote.Command, 0)
 	var parts []string
 
+	// Build hash beforehand as we need to pass it to the restart command
+	for filePath, fileDef := range integrations {
+		parts = append(parts, filePath, fileDef.Content)
+	}
+	for fullPath, fileDef := range files {
+		parts = append(parts, fullPath, fileDef.Content)
+	}
+	hash := utils.StrHash(parts...)
+
+	restartCmd, err := h.manager.restartAgentServices(func(name string, args command.Args) (string, command.Args) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
	// restart after integration file delete. To ensure the proper flow, we have the integration config file commands depending on this restart command. This way, when we delete the intg config command, pulumi will first delete the restart command calling the restartCmd.Delete. We transform the restart command to only define the `Delete` action
	restartCmd, err := h.manager.restartAgentServices(func(name string, args command.Args) (string, command.Args) {
```